src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala

```diff
@@ -22,6 +22,7 @@ class TaskIdUtil {
   def appId(taskId: String): PathId = {
     taskId match {
       case TaskIdRegex(appId, uuid) => PathId.fromSafePath(appId)
+      case _                        => PathId.empty
```

Comment:
Would it make sense to change the return type to `Try[PathId]` and return a `Failure` in case it can't extract a `PathId`?